docs/README.md

```diff
@@ -0,0 +1,148 @@
+
+#### Rollback
+One of the most compelling characteristics of "infrastructure-as-code" interfaces like the Kubernetes Resource Model is the ability to revert a configuration to a last known golden state. The default/canary pattern defines the entire current state, enabling GitOps rollback.
+
+* revert to previously safe commit
+* kubectl apply KFService.yaml
+
+It's possible to revert to a commit the contains `canary` in the spec. The state will be reconciled to rollback to a configuration that runs both `default` and `canary`. This is possible with the data model, but somewhat of an anti-pattern for safe rollbacks. We recommend reverting to KFService spec that contains only `default`.
+
+### Resources
+The default implementation will attempt to provide sane defaults for CPU and MEMORY. If more control is needed, we defer to [Specifying Resources in Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container).
+
+By default, a Service will be configured with requests and limits of:
+* 1 CPU
+* 2 GB MEM
+
+In addition to specifying nvidia/gpu count, users may target specific GPUs on specific platforms by using annotations (e.g. "cloud.google.com/gke-accelerator": "nvidia-tesla-t4"):
+
+### Replicas
+Though Service is intended to be a serverless data model that relies on pure auto-scaling to determine replica count, some use cases are not tolerant to the limitations of auto-scaling. It is possible to specify `spec.minReplicas` and `spec.maxReplicas`.
+
+Replicas will be surfaced as part of `status.canary` and `status.default`. It is important to note that `spec.minReplicas` and `spec.maxReplicas` apply to both `default` and `canary`. This may result in large resource usage during rollout if `spec.minReplicas` is set to a large value.
+
+```
+status:
+  default:
+    name: "my-model-endpoint-1"
+    replicas: 9
+    traffic: 90
+  canary 
+    name: "my-model-endpoint-2"
+    replicas: 1
+    traffic: 10
+```
+
+### Conditions
+Conditions provide realtime feedback to users on the underlying state of their deployment. As the feature set of Service grows, additional Conditions will be added to the set. We will use the convention of positive-true conditions; the resource is healthy if there are no conditions who's status is False. If no conditions are present on the resource (e.g. Ready is not present), the resource is not healthy. Only one condition for each type can be available.  
+
+| Type        | Description |
+| ----------- | ----------- | 
+| Ready                | Becomes True when everything is are ready. |
+| RevisionsReady       | Beomces True when the underlying revisions are ready. |
+| ContainersHealthy    | Becomes True when containers are successfully pulled, started, and pass readinessChecks. |
+| ResourcesProvisioned | Becomes True when minReplicas matches provisionedReplias. If provisioning error detected, reports False immediately. |
+
+### Tensorflow
+TODO Elaborate
```

Comment:
Will you add it in this PR?

Reply:
Sure -- we didn't know this before. Thanks for the reminder.